libs/core/src/lib/card/card.component.ts

```diff
@@ -0,0 +1,123 @@
+import {
+    Component,
+    OnInit,
+    Renderer2,
+    ElementRef,
+    ViewEncapsulation,
+    ChangeDetectionStrategy,
+    Input,
+    HostBinding
+} from '@angular/core';
+
+import { CLASS_NAME, CardType } from './constants';
+import { getCardModifierClassNameByCardType } from './utils';
+
+let cardId = 0;
+
+@Component({
+    selector: 'fd-card',
+    templateUrl: './card.component.html',
+    styleUrls: ['./card.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CardComponent implements OnInit {
+    /** Badge */
+    @Input() badge: string;
+
+    /**
+     * Whether to apply compact mode
+     */
+    @Input()
+    set compact(compact: boolean) {
+        this._setCompact(compact);
+    }
+    get compact(): boolean {
+        return this._compact;
+    }
+
+    /** Indicates when card should show a loader  */
+    @Input()
+    isLoading = false;
+
+    /**
+     * cardType can be 'object' | 'standard' | 'component' | 'analytical' | 'list' | 'table' | 'quickView' | 'linkList'
+     * to indicate what card's type it belongs to
+     */
+    @Input()
+    set cardType(cardType: CardType) {
+        this._setCardType(cardType);
+    }
+    get cardType(): CardType {
+        return this._cardType;
+    }
```

Comment:
Wouldn't be much easier for you to use `CssClassBuilder`?

Reply:
I did not use `CssClassBuilder` intentially cause as far as I know Frank does not like this approach and try to avoid it in the platform project. To be honest I'm not a big fun of this too cause this approach potentially might have some performance issues, sometimes can lead to not [obvious bugs](https://github.com/SAP/fundamental-ngx/pull/3328)

But this is not the `platform` project so I switched to `CssClassBuilder` interface.